src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt

```diff
@@ -398,6 +409,46 @@ class AdjudicationsService(
     }
   }
 
+  suspend fun deleteReferral(deleteEvent: ReferralEvent) {
+    val eventData: ReferralAdditionalInformation = deleteEvent.additionalInformation
+    val telemetryMap = deleteEvent.additionalInformation.toTelemetryMap()
+
+    runCatching {
+      val adjudicationMapping =
+        adjudicationMappingService.getMappingGivenChargeNumber(eventData.chargeNumber)
+          .also {
+            telemetryMap["adjudicationNumber"] = it.adjudicationNumber.toString()
+          }
+
+      val charge = adjudicationsApiService.getCharge(
+        eventData.chargeNumber,
+        eventData.prisonId,
+      )
+
+      val outcome = charge.reportedAdjudication.outcomes.firstOrNull()
+
+      // If no adjudication outcome exists - delete on NOMIS
+      // if outcome exists - update NOMIS with current state
```

Comment:
the deletion of a referral **outcome** is a rollback to a referral state, in the NOMIS world. 
The deletion of a referral is treated like a deletion